File: sdk/python/examples/create_and_run.py
from __future__ import annotations

import os

from openyak_sdk import OpenyakClient


def main() -> None:
    with OpenyakClient(base_url=os.environ["OPENYAK_BASE_URL"]) as client:
        thread = client.create_thread(
            model="claude-sonnet-4-6",
            allowed_tools=["bash"],
        )
        result = thread.run("PARITY_SCENARIO:bash_stdout_roundtrip")
        print(result.status)
        print(result.final_text)
        print(result.usage)


if __name__ == "__main__":
    main()

File: sdk/typescript/examples/create-and-run.ts
import { OpenyakClient } from "../src/index.js";

const baseUrl = process.env.OPENYAK_BASE_URL;
if (!baseUrl) {
  throw new Error("Set OPENYAK_BASE_URL to a running local `openyak server` address.");
}

const client = new OpenyakClient({ baseUrl });
const thread = await client.createThread({
  model: "claude-sonnet-4-6",
  allowedTools: ["bash"],
});

const result = await thread.run("PARITY_SCENARIO:bash_stdout_roundtrip");

console.log({
  threadId: result.threadId,
  runId: result.runId,
  status: result.status,
  finalText: result.finalText,
  usage: result.usage,
});
